x/stakeibc/keeper/lsm.go

```diff
@@ -186,6 +186,42 @@ func (k Keeper) ShouldCheckIfValidatorWasSlashed(
 	return oldInterval.LT(newInterval)
 }
 
+// Loops through all active host zones, grabs queued LSMTokenDeposits for that host
+// that are in status TRANSFER_QUEUE, and submits the IBC Transfer to the host
+func (k Keeper) TransferAllLSMDeposits(ctx sdk.Context) {
+	for _, hostZone := range k.GetAllActiveHostZone(ctx) {
+		// Ignore hosts that have not been successfully registered
+		if hostZone.DelegationIcaAddress == "" {
+			continue
+		}
+
+		// Submit an IBC transfer for all queued deposits
+		queuedDeposits := k.RecordsKeeper.GetLSMDepositsForHostZoneWithStatus(
+			ctx,
+			hostZone.ChainId,
+			recordstypes.LSMTokenDeposit_TRANSFER_QUEUE,
+		)
+		for _, deposit := range queuedDeposits {
+			k.RecordsKeeper.UpdateLSMTokenDepositStatus(ctx, deposit, recordstypes.LSMTokenDeposit_TRANSFER_IN_PROGRESS)
+
+			if err := k.RecordsKeeper.IBCTransferLSMToken(
+				ctx,
+				deposit,
+				hostZone.TransferChannelId,
+				hostZone.DepositAddress,
+				hostZone.DelegationIcaAddress,
+			); err != nil {
+				k.Logger(ctx).Error(fmt.Sprintf("Unable to submit IBC Transfer of LSMToken for %v%s on %s: %s",
+					deposit.Amount, deposit.Denom, hostZone.ChainId, err.Error()))
+				continue
```

Comment:
Should we set the LSMTokenDeposit to the TRANSFER_FAILED state if there is an error here?  This isn't a timeout it is an actual error

Reply:
It a transfer times out the first time but then actually has a problem, will it always continue and be retried each end blocker forever?

Reply:
I was pondering this as well. To clarify this is just if the _submission_ failed - meaning the transfer never got off the ground on Stride. The situation where we previously change the status to TRANSFER_FAILED is if the transfer left Stride successfully but failed on the hub. 

That said, I think this would only fail if either (a) the LSM token is no longer in the deposit account or (b) an invalid denom is specified. Either case is unlikely and would indicate a larger problem, so I'm actually thinking it might make sense to label these as FAILED as well

Reply:
Agreed that this would indicate a larger problem. Maybe we can just add an event emission so that indexing would alert contributors.